docs/proposals/external-ca.md

```diff
@@ -0,0 +1,112 @@
+# External Certificate Authority Proposal
+
+## Context
+
+To enable mTLS in the mesh, each dataplane has to receive a unique certificate. This certificate has to be signed by the root certificate which is unique for every mesh.
+Right now, Kuma auto-generates root certificate when mesh is created. Root certificates are stored in Postgres on Universal and as a Secret on Kubernetes.
+In many organisations there is already Certificate Authority, so there should be a way to use it with Kuma.
+
+## Requirements
+
+User wants to generate root certificate and provide it to Kuma.
+
+## Proposed configuration model
+
+```yaml
+type: Mesh
+name: default
+mtls:
+  ca:
+    external: {}
+  enabled: true
+```
+
+## How to provide certificates to Kuma
+
+There are several ways we can implement providing root certs for Kuma
+
+### File
+
+```yaml
+type: Mesh
+name: default
+mtls:
+  ca:
+    external:
+      keyPath: /path/to/file
+      certPath: /path/to/file
+  enabled: true
+```
+
+It seems simple, but storing key files on the machines with control plane is not ideal.
+The bigger problem though is that you have to synchronize files between CP instances. Once it's not in sync (Puppet fails, any other reason), the mesh will break.
+We already have storage to share between CP instances and it's Postgres on Universal and Etcd on K8S.
+  
+### HTTP Server + kumactl
+
+We can adopt the same pattern as with Dataplane Token Server.
```

Comment:
Let's explain why we need a separate HTTP(S) server instead of adding a new endpoint to the REST API.